### DISCUSSION TITLE
Custom rules after upgrading to v13

### DISCUSSION THREAD

**Discussion by ramarivera** [angular-eslint/angular-eslint#893] [General]
Hey team!!

Not sure if its the right place, but after updating to v13 my custom rules unit tests stopped working.
After checking the current version of the rules, I noticed that for example the `Element` selector now is `Element$1`. 

Having already solved my issue, now I am curious as to why that change in template parsing. Was it due to some other angular dependency being updated?

Thanks

**Comment by JamesHenry** [comment]
@ramarivera The Template AST that the Angular Compiler produces is not under our control, and is sadly subject to breaking changes at any time. This was one such breaking change they made and we had to update our rules accordingly when we discovered it.

**Reply by ramarivera** [reply]
Got it, thanks for confirming @JamesHenry!
Did they announce this breaking change somehow?
If not, could it be that next time it'd be announced in the migration issue for angular-ESLint? . I tried looking for it, but couldn't find any such mention 😔

**Reply by JamesHenry** [reply]
No the AST is not part of the public API, which is why it can change at any time, and they therefore do not publicize changes to it. This should hopefully change in the future now that they have started opening up APIs for writing compiler checks, but for now that's just how it is!

**Reply by ramarivera** [reply]
Oh that's nice to hear. Thanks for clarifying all that :D
